### ISSUE TITLE
tracker.rs doesn't log the slow request if it exceeds the deadline 

### ISSUE THREAD

**Issue by coderplay** [tikv/tikv#12088]
## Bug Report

If a coprocessor task takes longer than 1 minute, it can't be logged by tracker slowlog.

https://github.com/tikv/tikv/blob/master/src/coprocessor/tracker.rs#L256

### What did you expect?

We want to get more information from the slow log to see why it take more than 1 minute.

**Comment by youjiali1995** [comment]
Good catch. We will improve it.

**Comment by JayiceZ** [comment]
> Good catch. We will improve it.

Can I try to fix it?

**Comment by youjiali1995** [comment]
@JayiceZ Of course.

**Comment by sticnarf** [comment]
Per https://github.com/tikv/tikv/pull/12154#issuecomment-1067610392, personally I think it not a good idea to simply log the request as long as the request exceeds the deadline. This may cause log explosion under heavy load.

Instead, I'm thinking about returning enough information to TiDB and let TiDB aggregate and log the slow query. For example, how many requests are dropped due to exceeding the deadline, how much cpu time and i/o is wasted on these aborted executions.

**Comment by coderplay** [comment]
Agreed on the aggregated summary to avoid the log flooding. However, why not do the aggregation in-place on the current  TiKV node?  There are basically multiple advantages doing it on TiKV node 1) saving a networking communications between TiKV and TiDB 2) in case there is no TiDB, user just run TiKV  3) in case a single node is busy while the others are idle because of data skew/hotspot,  per TiKV node level aggregation should be more useful.

**Comment by sticnarf** [comment]
A node level aggregation can be done through the metrics. We will try improving the Grafana dashboards.

To aggregate query-level information, we have to return these statistics to TiDB, because a TiDB query consists of multiple coprocessor requests.

The tracker only works for the tidb_query coprocessor. It's not useful for KV users. Anyway, it is better to provide the similar statistics in KV interfaces.
